.github/workflows/sts-mock-push-to-main.yml

```diff
@@ -0,0 +1,250 @@
+          version: 1.123.0
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@55f725fcb586ca16a5ed5b0d75d464defbfa831b #main
+        with:
+          aws-region: eu-west-2
+          role-to-assume: ${{ secrets.STS_MOCK_DEV_GH_ACTIONS_ROLE_ARN }}
+
+      - name: Sam Validate
+        run: |
+          echo "SAM_CLI_TELEMETRY=0" >> $GITHUB_ENV
+          sam validate --lint
+
+      - name: Sam Build
+        run: |
+          sam build --cached
+
+      - name: Upload SAM artifact into the DEV artifact bucket
+        uses: govuk-one-login/devplatform-upload-action@6985ccbaa306e2320a8826252c922af65242b283 #main
+        with:
+          artifact-bucket-name: ${{ secrets.STS_MOCK_DEV_ARTIFACT_BUCKET }}
+          signing-profile-name: ${{ secrets.DEV_SIGNING_PROFILE_NAME }}
+          template-file: .aws-sam/build/template.yaml
+          working-directory: sts-mock
+
+  # build-and-push-test-image-to-build:
+  #   name: Build and push test image to Build
+  #   needs: 
+  #     - sts-mock-tests-and-sonar-scan
+  #     - build-and-push-test-image-to-dev
+  #   runs-on: ubuntu-latest
+  #   env:
+  #     STS_MOCK_BUILD_TEST_IMAGE_REPOSITORY_URI: ${{ secrets.STS_MOCK_BUILD_TEST_IMAGE_REPOSITORY_URI }}      
+  #     BUILD_CONTAINER_SIGN_KMS_KEY: ${{ secrets.BUILD_CONTAINER_SIGN_KMS_KEY }}
+  #     IMAGE_TAG: latest
+  #   defaults:
+  #     run:
+  #       shell: bash
+  #       working-directory: sts-mock
+  #   steps:
+  #   - name: Check out repository code
+  #     uses: actions/checkout@9a9194f87191a7e9055e3e9b95b8cfb13023bb08 #main
+  #     with:
+  #       submodules: true
+  #       fetch-depth: 0
+
+  #   - name: Build test image
+  #     run: |
+  #       docker build -t $STS_MOCK_BUILD_TEST_IMAGE_REPOSITORY_URI:$IMAGE_TAG . 
+
+  #   - name: Configure AWS credentials for BUILD
+  #     uses: aws-actions/configure-aws-credentials@55f725fcb586ca16a5ed5b0d75d464defbfa831b #main
+  #     with:
+  #       aws-region: eu-west-2
+  #       role-to-assume: ${{ secrets.STS_MOCK_BUILD_GH_ACTIONS_ROLE_ARN }}
+
+  #   - name: Login to Amazon ECR BUILD
+  #     uses: aws-actions/amazon-ecr-login@a81a5945e74802f35ca53aa274a9e00436e6210e #main
+
+  #   - name: Push image to BUILD
+  #     run: |
+  #       docker push $STS_MOCK_BUILD_TEST_IMAGE_REPOSITORY_URI:$IMAGE_TAG
+
+  #   - name: Install Cosign
+  #     uses: sigstore/cosign-installer@4959ce089c160fddf62f7b42464195ba1a56d382 #main
+  #     with:
+  #       cosign-release: 'v1.9.0'
+
+  #   - name: Code sign the Docker image
+  #     id: cosign-image             
+  #     run: |
+  #       cosign sign --key awskms:///$BUILD_CONTAINER_SIGN_KMS_KEY $STS_MOCK_BUILD_TEST_IMAGE_REPOSITORY_URI:$IMAGE_TAG
+
+  build-and-upload-sam-artifact-to-build:
+    name: Validate & upload S3 artifact to Build
+    runs-on: ubuntu-latest
+    needs:
+      # - build-and-push-test-image-to-build
+      - build-and-upload-sam-artifact-to-dev
```

Comment:
Thinking about this - Dev isn't on our route to live - should we consider deploying to build at the same time we deploy to dev?

Reply:
This is an already established pattern within Mobile Pod, although it isn't the route to live, I'm guessing this helps reduces GitHub Action minutes if a job fails.

Reply:
Resolved.